## 机制
异常被抛出时，如果没有在本作用域被捕获，就会造成栈展开，提前结束本作用域，并且将在本作用域中，之前已经初始化的对象全部析构（即调用他们的析构函数），依次往上面的调用层次复制，直到找到调用层次中有一层可以捕获这个异常为止。要是无法解决，程序会自动调用terminate()。
## 原因1
如果本析构函数没有可以捕获异常的条件，就会把此异常传递给调用析构函数的作用域（如果是全局变量，是自动被调用的），程序会崩掉x1。
## 原因2
如果在一个普通成员函数中抛出了异常，但是需要析构本作用域已经构造好的所有对象，如果恰好这些对象的析构函数中又抛异常了，就意味着同时出现了两个异常需要处理，C++没有提供这个能力，程序会崩掉x2。
## 非要在析构函数中抛怎么办
要么不抛，要么吞掉这个异常。